src/python/pants/help/help_formatter.py

```diff
@@ -34,7 +35,7 @@ def add_option(ohis, *, category=None):
                 # No need to repeat those in the advanced section.
                 title = f"{display_scope} options"
                 lines.append(self.maybe_green(f"{title}\n{'-' * len(title)}\n"))
-                lines.extend(hard_wrap(oshi.description))
+                lines.extend(hard_wrap(oshi.description, width=terminal_width()))
```

Comment:
Each call to `terminal_width` is a syscall, so it might be good to just call it once at the top.